adapters/repos/db/vector/flat/index.go

```diff
@@ -793,3 +793,34 @@ func ValidateUserConfigUpdate(initial, updated schemaConfig.VectorIndexConfig) e
 func (index *flat) AlreadyIndexed() uint64 {
 	return atomic.LoadUint64(&index.count)
 }
+
+func (index *flat) NewQueryVectorDistancer(queryVector []float32) common.QueryVectorDistancer {
+	var distFunc func(nodeID uint64) (float32, error)
+	switch index.compression {
+	case compressionBQ:
+		queryVecEncode := index.bq.Encode(queryVector)
+		distFunc = func(nodeID uint64) (float32, error) {
+			vec, err := index.bqCache.Get(context.Background(), nodeID)
+			if err != nil {
+				return 0, err
+			}
+			return index.bq.DistanceBetweenCompressedVectors(vec, queryVecEncode)
+		}
+	case compressionPQ:
+		// use uncompressed for now
+		fallthrough
+	default:
```

Comment:
**thought:** why we need `fallthrough` ? we can instead just let it use default adn a comment or  do
```
case compressionPQ:
default:
```

Reply:
copied that from somewhere else in this file - @trengrj is it ok like this?

Reply:
just double checked - the same is used at two other locations in this file, so I think it is fine - correct tme if I am wrong :)

Reply:
confirmed with Abdel